lib/api/controller/collection.js

```diff
@@ -214,26 +211,29 @@ class CollectionController extends NativeController {
    * @param {Request} request
    * @returns {Promise.<Object>}
    */
-  deleteSpecifications (request) {
-    const
-      { index, collection } = this.getIndexAndCollection(request),
-      specificationsId = `${index}#${collection}`;
+  async deleteSpecifications (request) {
+    const { index, collection } = this.getIndexAndCollection(request);
+    const specificationsId = `${index}#${collection}`;
 
-    return this.kuzzle.internalIndex.delete('validations', specificationsId)
-      .catch(error => {
-        if (error.status === 404) {
-          return {
-            acknowledged: true
-          };
-        }
+    try {
+      await this.kuzzle.internalIndex.delete('validations', specificationsId);
```

Comment:
The `internalIndex` is not loosely coupled?

Reply:
See the PR changelog: `kuzzle.internalIndex` is an instance of the `core/shared/Store` class, which is meant to be a universal factory class to access storage spaces (used in several parts of the code). Under the hood, it calls the storage module using events.